app/controllers/supplejack_api/stories_controller.rb

```diff
@@ -101,12 +101,16 @@ def user_not_authorized
     def create_story_record_views
       return unless log_request_for_metrics?
 
-      payload = JSON.parse(response.body)
-      log = payload['contents']&.map do |record|
-        { record_id: record['record_id'], display_collection: record['content']['display_collection'] }
+      payload = JSON.parse(response.body, symbolize_names: true)
+      return if payload[:contents].blank?
+
+      records = payload[:contents].map do |record|
+        { record_id: record[:record_id], display_collection: record[:content][:display_collection] }
       end
+      # remove invalid story items for the metrics, eg: user inputed text
+      records.select! { |record| record[:record_id].present? && record[:display_collection].present? }
```

Comment:
might be clearer to use reject! here, up to you though :). https://apidock.com/ruby/v2_5_5/Array/reject

Reply:
Fair call, I'll do it.

Reply:
Done